tests/reftests/admin-cache.unix.test

```diff
@@ -0,0 +1,413 @@
+### opam admin cache | unordered
+Done.
+### sh cache-analysi.sh
+ipsum : md5 : exists, link to sha512
+ipsum : sha256 : exists, link to sha512
+ipsum : sha512 : exists
+--
+dolor : md5 : exists, link to sha512
+dolor : sha256 : exists, link to sha512
+dolor : sha512 : exists
+--
+sit : md5 : exists, link to sha512
+sit : sha256 : exists, link to sha512
+sit : sha512 : exists
+--
+amet : md5 : exists, link to sha512
+amet : sha256 : exists, link to sha512
+amet : sha512 : exists
+### :::::::::::::::::::::::::::::::::::::::::::
+### :III: remove strongest hash and recompute :
+### :::::::::::::::::::::::::::::::::::::::::::
+### rm -rf cache
+### opam admin cache | unordered
+[ipsum.1] synchronised (file://${BASEDIR}/arch-ipsum.tgz)
+[dolor.1] synchronised (file://${BASEDIR}/arch-dolor.tgz)
+[sit.1] synchronised (file://${BASEDIR}/arch-sit.tgz)
+[amet.1] synchronised (file://${BASEDIR}/arch-amet.tgz)
+Done.
+### sh cache-analysi.sh
+ipsum : md5 : exists, link to sha512
+ipsum : sha256 : exists, link to sha512
+ipsum : sha512 : exists
+--
+dolor : md5 : exists, link to sha512
+dolor : sha256 : exists, link to sha512
+dolor : sha512 : exists
+--
+sit : md5 : exists, link to sha512
+sit : sha256 : exists, link to sha512
+sit : sha512 : exists
+--
+amet : md5 : exists, link to sha512
+amet : sha256 : exists, link to sha512
+amet : sha512 : exists
+### rm -rf cache/sha512
+### sh cache-analysi.sh
+ipsum : md5 : exists, link to sha512 (absent)
+ipsum : sha256 : exists, link to sha512 (absent)
+ipsum : sha512 : not found
+--
+dolor : md5 : exists, link to sha512 (absent)
+dolor : sha256 : exists, link to sha512 (absent)
+dolor : sha512 : not found
+--
+sit : md5 : exists, link to sha512 (absent)
+sit : sha256 : exists, link to sha512 (absent)
+sit : sha512 : not found
+--
+amet : md5 : exists, link to sha512 (absent)
+amet : sha256 : exists, link to sha512 (absent)
+amet : sha512 : not found
+### opam admin cache | unordered
+[dolor.1] synchronised (file://${BASEDIR}/arch-dolor.tgz)
+[ipsum.1] synchronised (file://${BASEDIR}/arch-ipsum.tgz)
+[sit.1] synchronised (file://${BASEDIR}/arch-sit.tgz)
+[amet.1] synchronised (file://${BASEDIR}/arch-amet.tgz)
+Done.
+### sh cache-analysi.sh
+ipsum : md5 : exists, link to sha512
+ipsum : sha256 : exists, link to sha512
+ipsum : sha512 : exists
+--
+dolor : md5 : exists, link to sha512
+dolor : sha256 : exists, link to sha512
+dolor : sha512 : exists
+--
+sit : md5 : exists, link to sha512
+sit : sha256 : exists, link to sha512
+sit : sha512 : exists
+--
+amet : md5 : exists, link to sha512
+amet : sha256 : exists, link to sha512
+amet : sha512 : exists
+### ::::::::::::::::::::::::::::::::::::
+### :IV: add a other hashes afterwards :
+### ::::::::::::::::::::::::::::::::::::
+### : ipsum : only md5 then adding sha512
+### : dolor : md5 & sha256 then adding sha512
+### : sit   : sha256 then adding md5 & sha512
+### <sedi.sh>
+regexp=$1
+file=$2
+sed "$regexp" $file > $file.tmp
+mv $file.tmp $file
```

Comment:
with its present use it's not a problem but if it's used togethr with ${BASEDIR} or whatever in the future it can become a problem and fly under the radar
```suggestion
sed "$regexp" "$file" > "$file.tmp"
mv "$file.tmp" "$file"
```